src/web/mjs/connectors/VizShonenJump.mjs

```diff
@@ -156,10 +176,57 @@ export default class VizShonenJump extends Connector {
         return volumes;
     }
 
+    async _getVizMangaChapters(manga) {
+        let auth = await this._getUserInfo();
+        const request = new Request(new URL(manga.id, this.url), this.requestOptions);
+        let data = await this.fetchDOM(request);
+
+        if ( data.innerText.includes('MANGA Plus by SHUEISHA') ) {
+            alert('This website is geolocked. It can only be accessed from the USA.\nYou may use MANGA Plus instead.', this.label, 'info');
+            return;
+        }
+
+        return [...data.querySelectorAll('div > a.o_chapter-container[data-target-url], tr.o_chapter td.ch-num-list-spacing a.o_chapter-container[data-target-url]')]
+            .filter(element => {
```

Comment:
About that filter , test try https://www.viz.com/vizmanga/chapters/fushigi-yugi-genbu-kaiden.

Chapter 1 is free, chapter 2 is free. Its ok
For chapter three i have ""javascript:void('join to read');" cuase i am not logged.

Since its not what you are testing ((/javascript:tryReadChapter/) it end up being displayed while it shouldnt .

Maybe doing something like that woudl be ok? 

```
            if(/javascript:tryReadChapter/i.test(element.href)) {
                    return auth.isVizManga;
                }
                
                if(/join to read/i.test(element.href)) {
                    return auth.isLoggedIn;
                }
```

Reply:
for logged in subscribed members, tryReadChapter is correct check as they see that only. for non subscribers or non-logged in, its 'join to read'.

added the second condition in flow so now both subscribers and non subscribers work